workbench-core/datasets/src/schemas/createDataSet.ts

```diff
@@ -12,10 +12,18 @@ const CreateDataSetSchema: Schema = {
   properties: {
     datasetName: { type: 'string' },
     path: { type: 'string' },
-    region: { type: 'string' }
+    description: { type: 'string' },
+    customMetadata: {
+      type: 'object',
+      properties: {
+        owningProjectId: { type: 'string' }
```

Comment:
the reference to "owningProjectId" does not belong here. If we define it here, the metadata is prescribed, not custom. It is fine to extend this within Service Workbench and define it there as it is metadata which is custom to the service workbench product or the projects component.

Reply:
There was already a suggestion to move schema to api level, but that was not finalized. I also feel that schema just does not belong to core. This is used for API validation and purely relates to SWB. We still can have any shape of metadata in dataset service though

Reply:
Can we move this file out of the `workbench-core/datasets`? This will be shipped as a standalone NPM separate for SWB. It should not contain a pure SWB construct.

Reply:
After giving this further though, I think we can move the schema to `swb-app`. We can put all of the schema under the schema folder. Once the request is received we should validate it's in the correct shape to be passed to the workbench core services.

Also remember that in the request we can call the field `owningProjectId`, but when that field is written to DDB it should be written as `owner`. This is so we can use the preexisting DDB index `getResourceByOwner`